core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1139,9 +1139,13 @@ object Log {
       throw exception(dir)
 
     val name: String =
-      if (dirName.endsWith(DeleteDirSuffix)) dirName.substring(0, dirName.indexOf('.'))
+      if (dirName.endsWith(DeleteDirSuffix)) dirName.substring(0, dirName.lastIndexOf('.'))
       else dirName
 
+    if (!name.contains('-')) {
```

Comment:
I trunk we also have `if (dirName.endsWith(DeleteDirSuffix) && !dirName.matches("^(\\S+)-(\\S+)\\.(\\S+)" + DeleteDirSuffix))`. We need to double-check if this check is still needed given that.